dataactbroker/handlers/fileHandler.py

```diff
@@ -244,6 +244,44 @@ def finalize(self, jobId=None):
             # Unexpected exception, this is a 500 server error
             return JsonResponse.error(e,StatusCode.INTERNAL_ERROR)
 
+    def checkSubmissionById(self, submission_id, file_type):
+        """ Check that submission exists and user has permission to it
+
+        Args:
+            submission_id:  ID of submission to check
+            file_type: file type that has been requested
+
+        Returns:
+            Tuple of boolean indicating whether submission has passed checks, and http response if not
+
+        """
+        error_occurred = False
+
+        try:
+            submission = self.interfaces.jobDb.getSubmissionById(submission_id)
+        except ResponseException as exc:
+            if isinstance(exc.wrappedException, NoResultFound):
+                # Submission does not exist, change to 400 in this case since route call specified a bad ID
+                exc.status = StatusCode.CLIENT_ERROR
+                message = "Submission does not exist"
+                error_occurred = True
+            else:
+                raise exc
+        try:
+            self.checkSubmissionPermission(submission)
```

Comment:
It seems like we have to do this check a lot throughout the file route code. Does it make sense to refactor to use some kind of decorator instead?